docs/reference/analysis/tokenizers/simplepattern-tokenizer.asciidoc

```diff
@@ -0,0 +1,117 @@
+[[analysis-simplepattern-tokenizer]]
+=== Simple Pattern Tokenizer
+
+experimental[]
+
+The `simplepattern` tokenizer uses a regular expression to capture matching
+text as terms. The set of regular expression features it supports is more
+limited than the <<analysis-pattern-tokenizer,`pattern`>> tokenizer, but the
+tokenization is generally faster.
+
+This tokenizer does not support splitting the input on a pattern match, unlike
+the <<analysis-pattern-tokenizer,`pattern`>> tokenizer. To split on pattern
+matches using the same restricted regular expression subset, see the
+<<analysis-simplepatternsplit-tokenizer,`simplepatternsplit`>> tokenizer.
+
+This tokenizer uses http://lucene.apache.org/core//6_5_1/core/org/apache/lucene/util/automaton/RegExp.html[Lucene regular expressions].
+For an explanation of the supported features and syntax, see <<regexp-syntax,Regular Expression Syntax>>.
+
+The default pattern is the empty string, which produces no terms. This
+tokenizer should always be configured with a non-default pattern.
+
+[WARNING]
+.Beware of Pathological Regular Expressions
+========================================
+
+A badly written regular expression could run very slowly or even throw a
+StackOverflowError and cause the node it is running on to exit suddenly.
```

Comment:
Yeah, I was on the fence about including that since you can still (as far as I can tell) write regexes that do a lot of backtracking with its feature set. If it will cause an exception at regex-compile time then it's probably best to take this out.

Earlier someone made a good point about how if there are too many admonition blocks, then people don't pay as much attention to them (and they're definitely more important in the places that use Java regexes).

For reference, this is what the Lucene code has to say about it

>// TODO: the matcher here is naive and does have N^2 adversarial cases that are unlikely to arise in practice, e.g. if the pattern is
// aaaaaaaaaab and the input is aaaaaaaaaaa, the work we do here is N^2 where N is the number of a's.  This is because on failing to match
// a token, we skip one character forward and try again.  A better approach would be to compile something like this regexp
// instead: .* | <pattern>, because that automaton would not "forget" all the as it had already seen, and would be a single pass
// through the input.